src/polyfill/RTCIceTransport.ts

```diff
@@ -1,73 +1,66 @@
+    onstatechange: globalThis.RTCIceTransport['onstatechange'];
 
-    constructor(init: { pc: RTCPeerConnection, extraFunctions }) {
+    constructor({ pc }: { pc: RTCPeerConnection }) {
         super();
-        this.#pc = init.pc;
-        this.#extraFunctions = init.extraFunctions;
+        this.#pc = pc;
 
         // forward peerConnection events
-        this.#pc.addEventListener('icegatheringstatechange', () => {
-            this.dispatchEvent(new Event('gatheringstatechange'));
+        pc.addEventListener('icegatheringstatechange', () => {
+            const e = new Event('gatheringstatechange')
+            this.dispatchEvent(e)
+            this.ongatheringstatechange?.(e)
         });
-        this.#pc.addEventListener('iceconnectionstatechange', () => {
-            this.dispatchEvent(new Event('statechange'));
-        });
-
-        // forward events to properties
-        this.addEventListener('gatheringstatechange', (e) => {
-            if (this.ongatheringstatechange) this.ongatheringstatechange(e);
-        });
-        this.addEventListener('statechange', (e) => {
-            if (this.onstatechange) this.onstatechange(e);
+        pc.addEventListener('iceconnectionstatechange', () => {
+            const e = new Event('statechange')
+            this.dispatchEvent(e)
+            this.onstatechange?.(e)
         });
     }
 
     get component(): globalThis.RTCIceComponent {
         const cp = this.getSelectedCandidatePair();
-        if (!cp) return null;
+        if (!cp?.local) return null;
         return cp.local.component;
     }
 
     get gatheringState(): globalThis.RTCIceGatheringState {
-        return this.#pc ? this.#pc.iceGatheringState : 'new';
+        return this.#pc.iceGatheringState;
     }
 
-    get role(): string {
+    get role(): globalThis.RTCIceRole {
         return this.#pc.localDescription.type == 'offer' ? 'controlling' : 'controlled';
     }
 
     get state(): globalThis.RTCIceTransportState {
-        return this.#pc ? this.#pc.iceConnectionState : 'new';
+        return this.#pc.iceConnectionState;
     }
 
     getLocalCandidates(): RTCIceCandidate[] {
-        return this.#pc ? this.#extraFunctions.localCandidates() : [];
+        return this.#pc.localCandidates;
     }
 
-    getLocalParameters(): any {
-        /** */
+    getLocalParameters(): RTCIceParameters | null {
+        return new RTCIceParameters(new RTCIceCandidate({ candidate: this.#pc.getSelectedCandidatePair().local.candidate, sdpMLineIndex: 0 }))
     }
 
     getRemoteCandidates(): RTCIceCandidate[] {
-        return this.#pc ? this.#extraFunctions.remoteCandidates() : [];
+        return this.#pc.remoteCandidates;
     }
 
-    getRemoteParameters(): any {
-        /** */
+    getRemoteParameters(): RTCIceParameters | null {
+        return new RTCIceParameters(new RTCIceCandidate({ candidate: this.#pc.getSelectedCandidatePair().remote.candidate, sdpMLineIndex: 0 }))
```

Comment:
RTCIceParams were completely missing, these still dont have password/ufrag, need to wait for upstream to expose that